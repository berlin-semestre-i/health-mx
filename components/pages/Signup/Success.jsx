import React from 'react'
import { Grid, List, Button, Container, Icon, Step, Checkbox } from 'semantic-ui-react'
import Link from 'next/link'
import styled, { createGlobalStyle } from 'styled-components'
import media from 'styled-media-query'

const Success = () => (
  <React.Fragment>
    <LoginBody />
    <LoginGrid>
      <ImageColumn className="image-container" width={9}/>
      <LoginColumn mobile={16} tablet={7} computer={7}>
        <LoginContainer>
          <SuccessIcon name="check circle" size="massive" />
          <h1 className="ui header">Solicitud enviada</h1>
          <InfoContent>
            <p>
            Gracias por enviar tu solicitud. En las próximas horas
            recibirás un correo electrónico con la aprobación de
            tu cuenta y contraseña temporal.
            </p>
          </InfoContent>
          <Link href="/"><Button secondary>Volver a pantalla de inicio</Button></Link>
        </LoginContainer>
      </LoginColumn>
    </LoginGrid>
  </React.Fragment>
)

export default Success

const LoginBody = createGlobalStyle`
  body {
    background-color: #FFFFFF;
  }
`
const LoginGrid = styled(Grid)`
  && {
    height: 100vh;
    margin: 0;
    margin-left: -84px;
  }
`
const ImageColumn = styled(Grid.Column)`
  ${media.greaterThan('large')`
    &&&.image-container {
      padding: 0;
      background-image: url("../../static/images/enfermera.png");
      background-repeat: no-repeat;
      background-size: 100%;
    }
  `}

  ${media.lessThan('large')`
    &&&.image-container {
      display: none;
    }
  `}
`
const LoginColumn = styled(Grid.Column)`
  & {
    margin: auto;
  }
`
const LoginContainer = styled(Container)`
  ${media.lessThan('large')`
    & {
      padding: 0 1rem;
    }
  `}

  ${media.greaterThan('large')`
    & {
      padding: 0 4.5rem;
    }
  `}

  & {
      text-align: center;
  }

  &&& .field, &&& .button {
    margin-bottom: 2em;
  }

  &&& .field label {
    font-size: 16px;
    font-weight: normal;
  }

  &&& .field label:after {
    display: none;
  }

  &&& img {
    max-width: 65%;
    margin-bottom: 3rem;
  }

  &&& .greet {
      margin-bottom: 2rem;
  }

  &&& .instructions{
      margin-bottom: 2rem;
  }

  &&& .input p{
      font-weight: 500;
  }
`
const SuccessIcon = styled(Icon)`
  & {
      color: #6DC00C;
  }
`

const InfoContent = styled.div`
  & {
      margin-top: 3rem;
      margin-bottom: 3rem;
  }
`