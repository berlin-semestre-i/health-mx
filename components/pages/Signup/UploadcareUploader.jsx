import React from 'react'
import { Button } from 'semantic-ui-react'
import uploadcare from 'uploadcare-widget'
import getConfig from 'next/config'

const config = getConfig()

const {UPLOADCARE_PUBLIC_KEY} = (config) ? config.publicRuntimeConfig : {}

const selectUploadCareImage = async ( onChange, onSuccessHandler ) => {
  uploadcare.start({ publicKey: UPLOADCARE_PUBLIC_KEY })

  try{
    const image = await uploadcare.openDialog( null, {
      crop: '',
      imagesOnly: true,
    } )
    onSuccessHandler(image.cdnUrl, onChange)
  } catch (error) {/*This prevents an error if user won't upload image*/}
}

const UploadcareUploader = ({label, onChange, onSuccessHandler, UPLOADCARE_PUBLIC_KEY }) => (
  <React.Fragment>
    <Button
      secondary
      type="button"
      onClick={() => selectUploadCareImage(onChange, onSuccessHandler, UPLOADCARE_PUBLIC_KEY)}
    >
      {label}
    </Button>
  </React.Fragment>
)

export default UploadcareUploader