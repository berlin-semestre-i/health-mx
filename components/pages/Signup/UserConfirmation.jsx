import React from 'react'
import { Grid, List, Button, Container, Icon, Step, Checkbox } from 'semantic-ui-react'
import Link from 'next/link'
import styled, { createGlobalStyle } from 'styled-components'
import media from 'styled-media-query'

export class UserConfirmation extends React.Component {
    continue = e => {
      e.preventDefault()
      this.props.nextStep()
    }

    back = e => {
      e.preventDefault()
      this.props.prevStep()
    }

    render(){
      const {
        values: { curp, nss, email },
      } = this.props

      return(
        <React.Fragment>
          <LoginBody />
          <LoginGrid>
            <ImageColumn className="image-container" width={9}/>
            <LoginColumn mobile={16} tablet={7} computer={7}>
              <LoginContainer>
                <Link href="">
                  <h3 className="ui header" onClick={this.back}>
                    <Icon fitted name="angle left" /> Atrás
                  </h3>
                </Link>
                <StepGroup size="mini">
                  <Step>
                    <Step.Title>Inicio</Step.Title>
                  </Step>
                  <Step>
                    <Step.Title>Validación</Step.Title>
                  </Step>
                  <Step>
                    <Step.Title>Fotografía</Step.Title>
                  </Step>
                  <Step active>
                    <Step.Title>Confirmación</Step.Title>
                  </Step>
                </StepGroup>
                <div className="instructions">
                  <p>
                Revisa que tus datos sean correctos y haz clic en <strong>Enviar solicitud </strong>
                para finalizar tu registro a Health-MX.
                  </p>
                </div>
                <List>
                  <ListItem>
                    <List.Content>
                      <List.Header>CURP</List.Header>
                      <List.Description>{curp}</List.Description>
                    </List.Content>
                  </ListItem>
                  <ListItem>
                    <List.Content>
                      <List.Header>NSS</List.Header>
                      <List.Description>{nss}</List.Description>
                    </List.Content>
                  </ListItem>
                  <ListItem>
                    <List.Content>
                      <List.Header>Correo electrónico</List.Header>
                      <List.Description>{email}</List.Description>
                    </List.Content>
                  </ListItem>
                </List>
                <InfoContent>
                  <Checkbox label={<label>He leído y acepto el
                    <Link href=""> aviso de privacidad</Link> y
                    <Link href=""> términos y condiciones</Link>.</label>}
                  />
                </InfoContent>
                <Link href="#">
                  <FinishButton
                    small="true"
                    primary type="submit"
                    onClick={this.continue}
                  >
                      Enviar solicitud
                  </FinishButton>
                </Link>
              </LoginContainer>
            </LoginColumn>
          </LoginGrid>
        </React.Fragment>
      )
    }
}

export default UserConfirmation

const LoginBody = createGlobalStyle`
  body {
    background-color: #FFFFFF;
  }
`
const LoginGrid = styled(Grid)`
  && {
    height: 100vh;
    margin: 0;
    margin-left: -84px;
  }
`
const ImageColumn = styled(Grid.Column)`
  ${media.greaterThan('large')`
    &&&.image-container {
      padding: 0;
      background-image: url("../../static/images/enfermera.png");
      background-repeat: no-repeat;
      background-size: 100%;
    }
  `}

  ${media.lessThan('large')`
    &&&.image-container {
      display: none;
    }
  `}
`
const LoginColumn = styled(Grid.Column)`
  & {
    margin-top: 9rem;
  }
`
const LoginContainer = styled(Container)`
  ${media.lessThan('large')`
    & {
      padding: 0 1rem;
    }
  `}

  ${media.greaterThan('large')`
    & {
      padding: 0 4.5rem;
    }
  `}

  &&& .field, &&& .button {
    margin-bottom: 2em;
  }

  &&& .field label {
    font-size: 16px;
    font-weight: normal;
  }

  &&& .field label:after {
    display: none;
  }

  &&& img {
    max-width: 65%;
    margin-bottom: 3rem;
  }

  &&& .greet {
      margin-bottom: 2rem;
  }

  &&& .instructions{
      margin-bottom: 2rem;
  }

  &&& .input p{
      font-weight: 500;
  }
`
const FinishButton = styled(Button)`
  & {
    width: 100%;
  }
`
const ListItem = styled(List.Item)`
  & {
      margin-bottom: 1rem;
  }
`
const InfoContent = styled.div`
  & {
    text-align: left;
    font-size: 14px;
    line-height: 3rem;
    margin-top: 3rem;
    margin-bottom: 4rem;
  }

  &.footer {
    margin-top: 5rem;
  }

  & a {
    font-weight: 500;
  }
`
const StepGroup = styled(Step.Group)`
  &.steps {
      border: none;
  }
  &.steps .step{
      border-right: none;
      color: #818EA3;
  }
  &.steps .step.active {
      background: transparent;
  }
  &.steps .step.active:after {
    background: transparent;
}
`