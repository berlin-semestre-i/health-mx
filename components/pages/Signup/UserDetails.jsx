import React from 'react'
import { Grid, Form, Button, Container, Icon, Step } from 'semantic-ui-react'
import Link from 'next/link'
import styled, { createGlobalStyle } from 'styled-components'
import media from 'styled-media-query'

export class UserDetails extends React.Component {
  continue = e => {
    e.preventDefault()
    this.props.nextStep()
  }

  render(){
    const { values, handleChange } = this.props
    return(
      <React.Fragment>
        <LoginBody />
        <LoginGrid>
          <ImageColumn className="image-container" width={9}/>
          <LoginColumn mobile={16} tablet={7} computer={7}>
            <LoginContainer>
              <Link href="/">
                <h3 className="ui header"><Icon fitted name="angle left" /> Iniciar sesión</h3>
              </Link>
              <StepGroup size="mini">
                <Step active>
                  <Step.Title>Inicio</Step.Title>
                </Step>
                <Step>
                  <Step.Title>Validación</Step.Title>
                </Step>
                <Step>
                  <Step.Title>Fotografía</Step.Title>
                </Step>
                <Step>
                  <Step.Title>Confirmación</Step.Title>
                </Step>
              </StepGroup>
              <div className="instructions">
                <p>
                    Para registrarte, es necesario contar con tu CURP, NSS, correo electrónico
                    e identificación oficial.
                </p>
              </div>
              <Form>
                <Form.Input
                  label="CURP"
                  placeholder="CURP"
                  required
                  maxLength="18"
                  onChange={handleChange('curp')}
                  defaultValue={values.curp}
                />
                <Form.Input
                  label="Número de Seguro Social (NSS)"
                  placeholder="NSS"
                  required
                  maxLength="11"
                  onChange={handleChange('nss')}
                  defaultValue={values.nss}
                />
                <Form.Input
                  label="Correo electrónico"
                  placeholder="john@ejemplo.com"
                  required
                  onChange={handleChange('email')}
                  defaultValue={values.email}
                />
                <Link href="#">
                  <NextButton
                    small="true"
                    primary type="submit"
                    onClick={this.continue}
                  >
                      Siguiente
                  </NextButton>
                </Link>
              </Form>
            </LoginContainer>
          </LoginColumn>
        </LoginGrid>
      </React.Fragment>
    )
  }

}

export default UserDetails

const LoginBody = createGlobalStyle`
  body {
    background-color: #FFFFFF;
  }
`
const LoginGrid = styled(Grid)`
  && {
    height: 100vh;
    margin: 0;
    margin-left: -84px;
  }
`
const ImageColumn = styled(Grid.Column)`
  ${media.greaterThan('large')`
    &&&.image-container {
      padding: 0;
      background-image: url("../../static/images/enfermera.png");
      background-repeat: no-repeat;
      background-size: 100%;
    }
  `}

  ${media.lessThan('large')`
    &&&.image-container {
      display: none;
    }
  `}
`
const LoginColumn = styled(Grid.Column)`
  & {
    margin-top: 9rem;
  }
`
const LoginContainer = styled(Container)`
  ${media.lessThan('large')`
    & {
      padding: 0 1rem;
    }
  `}

  ${media.greaterThan('large')`
    & {
      padding: 0 4.5rem;
    }
  `}

  &&& .field, &&& .button {
    margin-bottom: 2em;
  }

  &&& .field label {
    font-size: 16px;
    font-weight: normal;
  }

  &&& .field label:after {
    display: none;
  }

  &&& img {
    max-width: 65%;
    margin-bottom: 3rem;
  }

  &&& .instructions {
      margin-bottom: 2rem;
  }
`
const NextButton = styled(Button)`
  & {
    float: right;
  }
`
const StepGroup = styled(Step.Group)`
  &.steps {
      border: none;
  }
  &.steps .step{
      border-right: none;
      color: #818EA3;
  }
  &.steps .step.active {
      background: transparent;
  }
  &.steps .step.active:after {
    background: transparent;
}
`