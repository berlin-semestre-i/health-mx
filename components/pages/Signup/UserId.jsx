import React from 'react'
import { Grid, Form, Button, Container, Icon, Step } from 'semantic-ui-react'
import Link from 'next/link'
import styled, { createGlobalStyle } from 'styled-components'
import media from 'styled-media-query'
import UploadcareUploader from './UploadcareUploader'

export class UserId extends React.Component {
    continue = e => {
      e.preventDefault()
      this.props.nextStep()
    }

    back = e => {
      e.preventDefault()
      this.props.prevStep()
    }

    render(){
      const {
        values: { curp },
      } = this.props
      return(
        <React.Fragment>
          <LoginBody />
          <LoginGrid>
            <ImageColumn className="image-container" width={9}/>
            <LoginColumn mobile={16} tablet={7} computer={7}>
              <LoginContainer>
                <Link href="">
                  <h3 className="ui header" onClick={this.back}>
                    <Icon fitted name="angle left" /> Atrás
                  </h3>
                </Link>
                <StepGroup size="mini">
                  <Step>
                    <Step.Title>Inicio</Step.Title>
                  </Step>
                  <Step active>
                    <Step.Title>Validación</Step.Title>
                  </Step>
                  <Step>
                    <Step.Title>Fotografía</Step.Title>
                  </Step>
                  <Step>
                    <Step.Title>Confirmación</Step.Title>
                  </Step>
                </StepGroup>
                <div className="greet">
                  <p><strong>Hola, {curp}</strong></p>
                </div>
                <div className="instructions">
                  <p>
                Sube una imagen o documento de tu identificación oficial.
                Recuerda que la imagen o documento debe tener una calidad
                óptima para permitir la legibilidad.
                  </p>
                </div>
                <div className="input">
                  <p>Identificación oficial</p>
                  <UploadcareUploader mini="true" label="Subir archivo" />
                </div>
                <Link href="#">
                  <NextButton
                    small="true"
                    primary type="submit"
                    onClick={this.continue}
                  >
                      Siguiente
                  </NextButton>
                </Link>
              </LoginContainer>
            </LoginColumn>
          </LoginGrid>
        </React.Fragment>
      )
    }
}

export default UserId

const LoginBody = createGlobalStyle`
  body {
    background-color: #FFFFFF;
  }
`
const LoginGrid = styled(Grid)`
  && {
    height: 100vh;
    margin: 0;
    margin-left: -84px;
  }
`
const ImageColumn = styled(Grid.Column)`
  ${media.greaterThan('large')`
    &&&.image-container {
      padding: 0;
      background-image: url("../../static/images/enfermera.png");
      background-repeat: no-repeat;
      background-size: 100%;
    }
  `}

  ${media.lessThan('large')`
    &&&.image-container {
      display: none;
    }
  `}
`
const LoginColumn = styled(Grid.Column)`
  & {
    margin-top: 9rem;
  }
`
const LoginContainer = styled(Container)`
  ${media.lessThan('large')`
    & {
      padding: 0 1rem;
    }
  `}

  ${media.greaterThan('large')`
    & {
      padding: 0 4.5rem;
    }
  `}

  &&& .field, &&& .button {
    margin-bottom: 2em;
  }

  &&& .field label {
    font-size: 16px;
    font-weight: normal;
  }

  &&& .field label:after {
    display: none;
  }

  &&& img {
    max-width: 65%;
    margin-bottom: 3rem;
  }

  &&& .greet {
      margin-bottom: 2rem;
  }

  &&& .instructions{
      margin-bottom: 2rem;
  }

  &&& .input p{
      font-size: 16px;
      font-weight: 500;
  }
`
const NextButton = styled(Button)`
  & {
    float: right;
  }
`
const StepGroup = styled(Step.Group)`
  &.steps {
      border: none;
  }
  &.steps .step{
      border-right: none;
      color: #818EA3;
  }
  &.steps .step.active {
      background: transparent;
  }
  &.steps .step.active:after {
    background: transparent;
}
`