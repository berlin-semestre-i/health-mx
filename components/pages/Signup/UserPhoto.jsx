import React from 'react'
import { Grid, Form, Button, Container, Icon, Step } from 'semantic-ui-react'
import Link from 'next/link'
import styled, { createGlobalStyle } from 'styled-components'
import media from 'styled-media-query'
import UploadcareUploader from './UploadcareUploader'

export class UserPhoto extends React.Component {
    continue = e => {
      e.preventDefault()
      this.props.nextStep()
    }

    back = e => {
      e.preventDefault()
      this.props.prevStep()
    }

    render(){
      return(
        <React.Fragment>
          <LoginBody />
          <LoginGrid>
            <ImageColumn className="image-container" width={9}/>
            <LoginColumn mobile={16} tablet={7} computer={7}>
              <LoginContainer>
                <Link href="">
                  <h3 className="ui header" onClick={this.back}>
                    <Icon fitted name="angle left" /> Atrás
                  </h3>
                </Link>
                <StepGroup size="mini">
                  <Step>
                    <Step.Title>Inicio</Step.Title>
                  </Step>
                  <Step>
                    <Step.Title>Validación</Step.Title>
                  </Step>
                  <Step active>
                    <Step.Title>Fotografía</Step.Title>
                  </Step>
                  <Step>
                    <Step.Title>Confirmación</Step.Title>
                  </Step>
                </StepGroup>
                <div className="instructions">
                  <p>
                A continuación, sube una fotografía tuya mostrando la identificación
                que utilizaste en el paso anterior. <strong>Es importante que la fotografía
                muestre tu rostro de una forma clara.</strong>
                  </p>
                </div>
                <div className="input">
                  <p>Fotografía personal</p>
                  <UploadcareUploader mini="true" label="Subir archivo" />
                  <UploadcareUploader mini="true" label="Tomar fotografía" />
                </div>
                <Link href="#">
                  <NextButton
                    small="true"
                    primary type="submit"
                    onClick={this.continue}
                  >
                      Siguiente
                  </NextButton>
                </Link>
              </LoginContainer>
            </LoginColumn>
          </LoginGrid>
        </React.Fragment>
      )
    }
}

export default UserPhoto

const LoginBody = createGlobalStyle`
  body {
    background-color: #FFFFFF;
  }
`
const LoginGrid = styled(Grid)`
  && {
    height: 100vh;
    margin: 0;
    margin-left: -84px;
  }
`
const ImageColumn = styled(Grid.Column)`
  ${media.greaterThan('large')`
    &&&.image-container {
      padding: 0;
      background-image: url("../../static/images/enfermera.png");
      background-repeat: no-repeat;
      background-size: 100%;
    }
  `}

  ${media.lessThan('large')`
    &&&.image-container {
      display: none;
    }
  `}
`
const LoginColumn = styled(Grid.Column)`
  & {
    margin-top: 9rem;
  }
`
const LoginContainer = styled(Container)`
  ${media.lessThan('large')`
    & {
      padding: 0 1rem;
    }
  `}

  ${media.greaterThan('large')`
    & {
      padding: 0 4.5rem;
    }
  `}

  &&& .field, &&& .button {
    margin-bottom: 2em;
  }

  &&& .field label {
    font-size: 16px;
    font-weight: normal;
  }

  &&& .field label:after {
    display: none;
  }

  &&& img {
    max-width: 65%;
    margin-bottom: 3rem;
  }

  &&& .greet {
      margin-bottom: 2rem;
  }

  &&& .instructions{
      margin-bottom: 2rem;
  }

  &&& .input p{
      font-size: 16px;
      font-weight: 500;
  }
`
const NextButton = styled(Button)`
  & {
    float: right;
  }
`
const StepGroup = styled(Step.Group)`
  &.steps {
      border: none;
  }
  &.steps .step{
      border-right: none;
      color: #818EA3;
  }
  &.steps .step.active {
      background: transparent;
  }
  &.steps .step.active:after {
    background: transparent;
}
`