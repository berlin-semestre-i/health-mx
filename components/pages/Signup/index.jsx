import React from 'react'
import UserDetails from './UserDetails'
import UserId from './UserId'
import UserPhoto from './UserPhoto'
import UserConfirmation from './UserConfirmation'
import Success from './Success'

export class SignUp extends React.Component {
  state = {
    step: 1,
    curp: '',
    nss: '',
    email: '',
    id: '',
    photo: '',
  }

  nextStep = () => {
    const { step } = this.state
    this.setState({
      step: step + 1,
    })
  }

  prevStep = () => {
    const { step } = this.state
    this.setState({
      step: step - 1,
    })
  }

  handleChange = input => e => {
    this.setState({[input]: e.target.value})
  }

  render() {
    const { step } = this.state
    const { curp, nss, email, id, photo } = this.state
    const values = { curp, nss, email, id, photo }

    switch(step){
    case 1:
      return(
        <UserDetails
          nextStep={this.nextStep}
          handleChange={this.handleChange}
          values={values}
        />
      )
    case 2:
      return(
        <UserId
          nextStep={this.nextStep}
          prevStep={this.prevStep}
          handleChange={this.handleChange}
          values={values}
        />
      )
    case 3:
      return(
        <UserPhoto
          nextStep={this.nextStep}
          prevStep={this.prevStep}
          handleChange={this.handleChange}
          values={values}
        />
      )
    case 4:
      return(
        <UserConfirmation
          nextStep={this.nextStep}
          prevStep={this.prevStep}
          handleChange={this.handleChange}
          values={values}
        />
      )
    case 5:
      return(
        <Success/>
      )
    }
  }
}

export default SignUp
