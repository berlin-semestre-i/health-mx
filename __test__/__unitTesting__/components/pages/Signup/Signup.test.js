import React from 'react'
import { shallow } from 'enzyme'
import Signup from '../../../../../components/pages/Signup/index'

describe('Signup', () => {
  it('renders', () => {
    const component = shallow(<Signup />)
    expect(component).toMatchSnapshot()
  })
})