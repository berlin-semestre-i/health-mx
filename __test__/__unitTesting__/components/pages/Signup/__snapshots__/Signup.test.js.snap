// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup renders 1`] = `
<UserDetails
  handleChange={[Function]}
  nextStep={[Function]}
  values={
    Object {
      "curp": "",
      "email": "",
      "id": "",
      "nss": "",
      "photo": "",
    }
  }
/>
`;
